doc/api/process.md

```diff
@@ -1881,6 +1881,229 @@ a code.
 Specifying a code to [`process.exit(code)`][`process.exit()`] will override any
 previous setting of `process.exitCode`.
 
+## `process.finalization.register(ref, callback)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active Development
+
+* `ref` {Object | Function} The reference to the resource that is being tracked.
+* `callback` {Function} The callback function to be called when the resource
+  is finalized.
+  * `ref` {Object | Function} The reference to the resource that is being tracked.
+  * `event` {string} The event that triggered the finalization. Defaults to 'exit'.
+
+This function registers a callback to be called when the process emits the `exit` event.
+
+If the object `ref` is garbage collected before the `exit` event is emitted, the callback will be
+removed from the finalization registry, and it will not be called on process exit.
+
+Inside the callback you can do whatever you want with the object,
```

Comment:
This paragraph definitely can be improved, so let me try:

```suggestion
Inside the callback, you can release the resources allocated to that object. 

Also, this callback is called during `exit` event, so be aware that all the limitations of `exit` event are also applied in this callback, such as not being able to use async code since the work queue will not be called.
```

> It also contains a lot of assumptions about how the GC works despite a weak disclaimer about how unreliable GC is at the end. That might be fine for a user-land investigation but being in core it needs more proof-reading from V8 GC people.

Do you have suggestion of people that we can ping?

> It’s also not guaranteed that we will call into JS before exit during exceptional circumstances.

Inside the "Event: 'exit'" docs, there is no explicit mention of exceptional circumstances, if we follow the current docs, the assumption is the `exit` will always be called.

Reply:
> Do you have suggestion of people that we can ping?

I think it would be worth asking in the v8-user mailing list or ask in their issue tracker.